### PR TITLE
SONARJAVA-1928 FP on S1172 when owner of the method extends Struts 1.…

### PR DESCRIPTION
**Pull request by fergarrui**
* Avoiding a FP on S1172 (Unused Method Parameter Check). When a class is a subclass of `org.apache.struts.action.Action` (Struts 1.x) it has to add mandatory extra parameters.

### CODE REVIEW

**Comment by fergarrui** [comment]
Added another commit to check Action and its sub-classes.

**Comment by m-g-sonar** [comment]
Hello @fergarrui,
Thanks for your contribution. This JIRA ticket is planned for version 4.4 of the Java Analyzer. So I'll review in details after release of 4.3!

Cheers,
Michael

File: java-checks/src/test/files/checks/unused/UnusedMethodParameterCheck.java [hunk 1]
```diff
@@ -153,4 +159,32 @@ void unknownWarningCombinedWithKnown(List list, int unused) { // Compliant
...
+    System.out.println(""); 
+  }
+}
+
+class StrutsAction2 extends BaseAction {
+  void foo(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response, String unused) { // Noncompliant {{Remove this unused method parameter "unused".}}
+    System.out.println(""); 
+  }
+  
+  void bar(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) { // Compliant
```
Comment (m-g-sonar) [inline java-checks/src/test/files/checks/unused/UnusedMethodParameterCheck.java]: In order to completely cover you cases, add a method in a 3rd class which does not extends Action, having an unused parameter being from one of the struts types. This should raise an issue.

File: java-checks/src/main/java/org/sonar/java/checks/unused/UnusedMethodParameterCheck.java [hunk 2]
```diff
@@ -41,10 +45,6 @@
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.VariableTree;
 
-import java.util.ArrayList;
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/unused/UnusedMethodParameterCheck.java]: Please avoid changing the import orders.

**Comment by fergarrui** [comment]
Changes done:
* Imports reorder reverted
* Variable renamed (ouch!)
* Added new tests

**Comment by m-g-sonar** [comment]
Great! Thanks for the changes. Everything looks good to me now! I'll merge it once we released version 4.3.

Thanks again for your contribution!

Cheers,
Michael

**Pull request merged**